src/strategy/protobuf/index.ts

```diff
@@ -1,66 +1,35 @@
-import { Stream } from 'stream';
-import { ProtobufTransform } from '../../utils/protobuf-transform';
-import { TransformStrategy } from '../../types';
 import { ProtobufOptions } from './types/';
 import { load, Root, Type } from 'protobufjs';
-import { createGunzip, createGzip } from 'zlib';
 import { SerializerStrategy } from '../serializer';
 
-export class ProtobufStrategy implements SerializerStrategy {
-	private options!: ProtobufOptions;
-	private context!: Promise<Type>;
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export class ProtobufStrategy<A = any>
+	implements SerializerStrategy<A, Uint8Array>
+{
+	private type: Promise<Type>;
 
 	constructor(options: ProtobufOptions) {
-		this.options = options;
-		this.context = this.load(options);
+		this.type = this.load(options);
 	}
 
 	private async load(options: ProtobufOptions): Promise<Type> {
-		const type: Type | undefined = await new Promise(async (resolve) => {
-			await load(options.proto, (err: Error | null, root?: Root) => {
-				if (err) {
-					throw err;
-				}
-				resolve(root?.lookupType(options.attribute));
-			});
-		});
+		const type: Type | undefined = await new Promise(async (resolve, reject) =>
+			load(options.proto, (err: Error | null, root?: Root) =>
+				err ? reject(err) : resolve(root?.lookupType(options.attribute)),
+			),
+		);
```

Comment:
The throw err would be an untreated error and could shut down the service. Using reject instead gives an opportunity for the user to catch and treat the error if he wants to